### ISSUE TITLE
is it possible to combineReducers? (combine stores in restate terminology)

### ISSUE THREAD

**Issue by niron1** [endenwer/svelte-restate#1]

**Comment by endenwer** [comment]
The idea is to have only one store. Technically it is possible to combine data from multiple stores, but I would't do so. 
``` typescript
const sub1 = store1.regRootSub(
  'sub1',
  ($root) => $root.data1
)

const sub2 = store2.regRootSub(
  'sub2',
  ($root) => $root.data2
)

const sub3 = store1.regSub(
  'sub3',
  () => [sub1(), sub2()]
  ($sub1, $sub2) => $sub1 + $sub2
)
```

**Comment by niron1** [comment]
so is it not possible to create hierarchy as in redux and immutableJS?  or, do all entries has to be on the root level ? 🤔
as per an example, with immutableJS, you can do getIn(['level1', 'level2', 'level3']) to access local component's data
anyway, I also tried to run your example here, it doesn't seem to work.
if I understand correctly, you are suggesting to combine selectors, not stores.

**Comment by endenwer** [comment]
Yes, you need to combine selectors(subs here). If you want,  you can read more about this approach here https://day8.github.io/re-frame/subscriptions/
And the demo is running here https://svelte-restate-example.netlify.app/ You can look in Redux dev tools how it all works.

**Comment by niron1** [comment]
thanks, I am already using your library as is :)
I am currently bypassing the problem by creating objects hierarchy like so:

`export const getIbcTechnician = appDB.regRootSub(
    'ibcTechnician',
    ($state) => $state[ORDER_FORM_STATE].ibcTechnician
)`

`export const getIsErrorMessageDisplayed = appDB.regRootSub(
    'isErrorMessageDisplayed',
    ($state) => $state[ERROR_STATE].isErrorMessageDisplayed
)
`
(each constant ending with _STATE) represents a different category)
I am not sure how efficient is data access now, but it's currently more important to me to have the data arranged by categories and not flat.
I might come to a point to suggest a PR to combine multiple stores into one tree like CombineReducers if that's of interest.
but for the most part Restate resolved my main pain  - lack of immutability and redux-dev-tools integration. thanks

**Comment by endenwer** [comment]
I would do it like this
```typescript
export const orderForm = appDB.regRootSub(
  'orderForm',
  ($root) => $state[ORDER_FORM_STATE]
)

export const getIbcTechnician = appDB.regSub(
  'ibcTechnician',
  () => orderForm(),
  ($orderForm) => orderForm.ibcTechnician
)

export const errorState = appDB.regRootSub(
  'errorState',
  ($root) => $state[ERROR_STATE]
)

export const getIsErrorMessageDisplayed = appDB.regSub(
  'isErrorMessageDisplayed',
  () => errorState(),
  ($errorState) => errorState.isErrorMessageDisplayed
)
```
I don't plan to add new functionality. It is already working I as planned, but thanks for the suggestion.

**Comment by niron1** [comment]
not sure how this line could word:

````
export const orderForm = appDB.regRootSub(
  'orderForm',
  ($root) => $state[ORDER_FORM_STATE]
)
````

when the function input is $root and the usage is $state ?

**Comment by endenwer** [comment]
yes, that is a mistake, it should be like this.
```typescript
($root) => $root[ORDER_FORM_STATE]
```
